samples/src/Microsoft.Azure.IIoT.App/src/Pages/Discoverers.razor

```diff
@@ -151,50 +157,67 @@
     }
 
     /// <summary>
-    /// OnInitializedAsync
+    /// OnInitialized
     /// </summary>
-    protected override async Task OnInitializedAsync()
+    protected override void OnInitialized()
     {
-        discovererList = await RegistryHelper.GetDiscovererListAsync();
-        Page = "1";
-        pagedDiscovererList = discovererList.GetPaged(Int32.Parse(Page), PageLength);
-        _discovererEvent = await RegistryServiceEvents.SubscribeDiscovererEventsAsync(null, DiscovererEvent);
+        _spinner = "loader-big";
+    }
+
+    /// <summary>
+    /// OnAfterRenderAsync
+    /// </summary>
+    /// <param name="firstRender"></param>
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            discovererList = await RegistryHelper.GetDiscovererListAsync();
```

Comment:
Let's use query with paging.

Reply:
I'm already using query with paging

Reply:
So where are you calling the API to get the next page?   I can see you use query, but what I am missing is where you query again for the next page.   We should use the paging the API provides, not query all and then page the list, the former is far more efficient - also for all the other API, including browsing.